### PR TITLE
Add the Poll entity

### PR DESCRIPTION
**Pull request by cmrd-senya**

### CODE REVIEW

File: lib/diaspora_federation/validators/poll_answer_validator.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module DiasporaFederation
+  module Validators
+    class PollAnswerValidator < Validation::Validator
+      include Validation
+
+      rule :guid, :guid
+      rule :answer, :not_empty
```
Comment (SuperTux88) [inline lib/diaspora_federation/validators/poll_answer_validator.rb]: I think there is also a max length for this.

File: lib/diaspora_federation/validators/poll_validator.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module DiasporaFederation
+  module Validators
+    class PollValidator < Validation::Validator
+      include Validation
+
+      rule :guid, :guid
+      rule :question, :not_empty
```
Comment (SuperTux88) [inline lib/diaspora_federation/validators/poll_validator.rb]: max length here too.

**Comment by SuperTux88** [comment]
I think you should also add the PollParticipation

**Comment by cmrd-senya** [comment]
I don't see any length restrictions in [poll](https://github.com/diaspora/diaspora/blob/develop/app/models/poll.rb#L16) and [poll_answer](https://github.com/diaspora/diaspora/blob/develop/app/models/poll_answer.rb#L11) in opposite to [comment](https://github.com/diaspora/diaspora/blob/develop/app/models/comment.rb#L30)

**Comment by SuperTux88** [comment]
They have a limit of 255 at the database: https://github.com/diaspora/diaspora/blob/develop/db/schema.rb#L327

File: spec/lib/diaspora_federation/validators/poll_answer_validator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    describe "#guid" do
+      it_behaves_like "a guid validator" do
+        let(:property) { :guid }
+      end
+    end
+
+    describe "#answer" do
+      it_behaves_like "a property with a value validation/restriction" do
+        let(:property) { :answer }
+        let(:wrong_values) { [nil, "", "a"*256] }
```
Comment (diaspora-code-review) [inline spec/lib/diaspora_federation/validators/poll_answer_validator_spec.rb]: Surrounding space missing for operator `*`.

File: spec/lib/diaspora_federation/validators/poll_answer_validator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        let(:correct_values) { ["a"*255] }
```
Comment (diaspora-code-review) [inline spec/lib/diaspora_federation/validators/poll_answer_validator_spec.rb]: Surrounding space missing for operator `*`.

**Comment by cmrd-senya** [comment]
Restriction applied

File: spec/factories.rb [hunk 5]
```diff
@@ -5,6 +5,7 @@ def r_str
 end
 
 FactoryGirl.define do
+  initialize_with { new(attributes) }
```
Comment (SuperTux88) [inline spec/factories.rb]: didn't know this :+1:

**Comment by SuperTux88** [comment]
Thanks :)

File: spec/factories.rb (hunk 5 above)
Comment (cmrd-senya) [inline spec/factories.rb]: I found description [here](http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md#Custom_Construction), especially:

>You can override the initializer in order to:
>- Build non-ActiveRecord objects that require arguments to initialize

**Comment by cmrd-senya** [comment]
You are welcome :)

**Pull request merged**